static/src/javascripts/projects/common/modules/commercial/dfp/render-advert.js

```diff
@@ -128,14 +130,25 @@ define([
         removePlaceholders(advert.node);
 
         return applyCreativeTemplate(advert.node).then(function (isRendered) {
-            renderAdvertLabel(advert.node);
+            return renderAdvertLabel(advert.node)
+                .then(callSizeCallback)
+                .then(addRenderedClass)
+                .then(function () {
+                    return isRendered;
+                });
 
-            var size = slotRenderEvent.size.join(',');
-            if (sizeCallbacks[size]) {
-                sizeCallbacks[size](slotRenderEvent, advert);
+            function callSizeCallback() {
+                var size = slotRenderEvent.size.join(',');
+                if (sizeCallbacks[size]) {
+                    sizeCallbacks[size](slotRenderEvent, advert);
+                }
             }
 
-            return isRendered;
+            function addRenderedClass() {
+                return isRendered ? fastdom.write(function () {
+                    advert.node.classList.add('ad-slot--rendered');
```

Comment:
I'm happy to do one or the other, but not both, which I fear this approach is leading us to. I'll raise it in the eng meeting to gather the thoughts of the rest of the team. Hi Jimmy!